src/components/OpenWith/OpenWithButton.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @flow
+ * @file Preview Open With button
+ * @author Box
+ */
+
+import * as React from 'react';
+import Button from 'box-react-ui/lib/components/button/Button';
+import { injectIntl, FormattedMessage } from 'react-intl';
+import WithTooltip from './WithTooltip';
+import ICON_FILE_MAP from './IconFileMap';
+import messages from '../messages';
+
+type Props = {
+    extension: ?string,
+    displayIntegration: ?Integration,
+    onClick: ?Function,
+    tooltipText?: string | Object,
+    isDisabled: boolean,
+    isLoading: boolean,
+    icon?: string,
+};
+
+const INTEGRATION_ICON_CLASS = 'integration-icon';
+
+const OpenWithButton = ({
+    extension,
+    onClick,
+    displayIntegration,
+    tooltipText,
+    isDisabled = false,
+    isLoading = true,
+}: Props) => {
+    const { displayName = null, appIntegrationId = null } =
+        displayIntegration || {};
+    let DisplayIcon = ICON_FILE_MAP[displayName || 'default'];
```

Comment:
Make sure to handle a case where we have a display name that doesn't map to a known key.